Homework-16

StetsonCapital opened the pull request:
Registration Navigation Test

StetsonCapital committed:
Homework-16

Registration Navigation Test

duykhanhrc commented on src/test/java/Homework16.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.Duration;
+
+public class Homework16 extends BaseTest {
+  @test
```
Hi,
Your homework 16 has been checked, but the test script not run because you use wrong TestNG Annotations.

TestNG Annotations are used to control the next method to be executed in the test script. TestNG annotations are defined before every method in the test code. In case any method is not prefixed with annotations, it will be ignored and not be executed as part of the test code. To define them, methods need to be simply annotated with ‘@Test‘.

//HINT: you need to add `import org.testng.annotations.Test;` and use `@Test` annotations not `@test`

Please correct it.
Best wishes from the HW team (Khanh, khanh@testpro.io)

StetsonCapital replied:
Hello,
I have corrected the work and I added the class in Gradle and was able to get it to run successfully. THe only problem I have now is that when I run it, koel will open and close quickly. Isn't koel supposed to stay open when you run test?

duykhanhrc commented:
![image](https://github.com/StetsonCapital/A49-QA-Automation-Framework/assets/44592718/37f17c4f-cacd-4e17-8531-b8d06aec8641)

the useTestNG() method in the build.gradle is used to configure TestNG options for the default test task. If you run Homework16, you need to class `<class name="Homework16"/>` in the TestNG.xml file to run script

duykhanhrc commented on src/test/java/Homework16.java:
```diff
@@ -0,0 +1,32 @@
+      String registrationUrl = "https://qa.koel.app/registration";
+      Assert.assertEquals(driver.getCurrentUrl(), registrationUrl);
+
+      driver.quit();
```
Hi,
I have checked your homework 16, but I not see you resolve this error as I mention in the previous comment. (Issue about @Test testng annotations & add className in the suites TestNG.xml)
Please update code, push to your github and then submit link pull request for us review, run check again to complete this task.

![image](https://github.com/StetsonCapital/A49-QA-Automation-Framework/assets/44592718/c095125c-6ae2-48d0-98c8-f85e747d3791)


About your question `THe only problem I have now is that when I run it, koel will open and close quickly. Isn't koel supposed to stay open when you run test?`. At first step in line 14 to 21 to initialization chrome driver, open browser and get koel url. After that will come the steps to execute the action on the web. And at end step `driver.quit(); `browser will quit.

So browser always to open in the you run test.

By the way, selenium also support running test in headless mode (Headless browser is a program that emulates a browser but has no user interface)
You can use `options.addArguments("--remote-allow-origins=*");` or `options.setHeadless(true);` to run in the headless mode

StetsonCapital commented:
Hello sir,I did fix the @test problem. I changed it to @Test and it worked fine. I also added the correct class to build gradle.The only problem now is it not staying open which I think is because of drive.quit Sent from my iPhoneOn Aug 22, 2023, at 10:35 AM, Pham Duy Khanh ***@***.***> wrote:﻿
@duykhanhrc commented on this pull request.



In src/test/java/Homework16.java:
> +      ChromeOptions options = new ChromeOptions();
+      options.addArguments("--remote-allow-origins=*");
+
+      WebDriver driver = new ChromeDriver(options);
+      driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+
+      String url = "https://qa.koel.app/";
+      driver.get(url);
+
+      WebElement registrationLink = driver.findElement(By.cssSelector("[href='registration']"));
+      registrationLink.click();
+
+      String registrationUrl = "https://qa.koel.app/registration";
+      Assert.assertEquals(driver.getCurrentUrl(), registrationUrl);
+
+      driver.quit();

Hi,
I have checked your homework 16, but I not see you resolve this error as I mention in the previous comment. (Issue about @test testng annotations & add className in the suites TestNG.xml)
Please update code, push to your github and then submit link pull request for us review, run check again to complete this task.
About your question THe only problem I have now is that when I run it, koel will open and close quickly. Isn't koel supposed to stay open when you run test?. At first step in line 14 to 21 to initialization chrome driver, open browser and get koel url. After that will come the steps to execute the action on the web. And at end step driver.quit(); browser will quit.
So browser always to open in the you run test.
By the way, selenium also support running test in headless mode (Headless browser is a program that emulates a browser but has no user interface)
You can use options.addArguments("--remote-allow-origins=*"); or options.setHeadless(true); to run in the headless mode

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you modified the open/close state.Message ID: ***@***.***>
